### PR TITLE
chore: Update apk generation and naming convention

### LINKED ISSUE #2292: Update APK generation and follow name conventions
Please reduce the number of APKs generated and change the generation of the APK file as follows:

- susiai-dev-debug.apk
- susiai-dev-debug-fdroid.apk
- susiai-master-debug.apk
- susiai-master-debug-fdroid.apk
- susiai-master-unsigned.apk
- susiai-master-unsigned-fdroid.apk

### PR DESCRIPTION
**Pull request by atm1504**
Fixes #2292

### CODE REVIEW

**Comment by atm1504** [comment]
@iamareebjamal I tried to fix the issue by going through the previous PRs and the existing code in the open event app. And by going through the blog in fossasia blog.
 So, there might be wrong. But I think it would work. Please review accordingly.

**Comment by iamareebjamal** [comment]
You don't need to touch anything above, just call rm on unneeded files and change the naming in mv

**Comment by atm1504** [comment]
@iamareebjamal please review it now

File: exec/apk-gen.sh [hunk 1]
```diff
@@ -37,6 +37,28 @@ then
 		done
     	fi
     	# Create a new branch that will contain only latest apk
+
+    # Remove unwanted apk files
+    rm app-fdroid-release-unsigned.apk
+    rm app-playStore-release-unaligned.apk
+    rm app-playStore-release-unsigned.apk
```
Comment (iamareebjamal) [inline exec/apk-gen.sh]: Need this

File: exec/apk-gen.sh [hunk 2]
```diff
@@ -37,6 +37,28 @@ then
...
+    rm susi-ai-master-app-playStore-release-unaligned.apk
+    rm susi-ai-master-app-playStore-release.apk
```
Comment (iamareebjamal) [inline exec/apk-gen.sh]: Need this

**Comment by iamareebjamal** [comment]
Your script will fail. Seems you haven't tested it

**Comment by atm1504** [comment]
> Your script will fail. Seems you haven't tested it

Tried testing with the following code snippet, but didn't find any response in the terminal both before and after making changes.
```
chmod +x exec/apk-gen.sh
sh exec/apk-gen.sh

**Comment by atm1504** [comment]
> Your script will fail. Seems you haven't tested it

@iamareebjamal  I tested the script for the fdroid apps (as the local builds are fdroid only), it worked fine. It was able to rename and delete the files also.

File: exec/apk-gen.sh [hunk 3]
```diff
@@ -37,6 +37,28 @@ then
...
+
+    # Rename files accordingly
+    mv app-fdroid-debug.apk susiai-dev-debug-fdroid.apk
+    mv app-playStore-debug.apk susiai-dev-debug.apk
+    mv susi-ai-master-app-fdroid-debug.apk susiai-master-debug-fdroid.apk
+    mv susi-ai-master-app-playStore-debug.apk susiai-master-debug.apk
+    mv susi-ai-master-app-fdroid-release-unsigned.apk susiai-master-unsigned-fdroid.apk
+    mv susi-ai-master-app-playStore-release-unsigned.apk susiai-master-unsigned.apk
+    mv app-playStore-release-unsigned.apk susiai-dev-release-unsigned.apk
+    mv susi-ai-master-app-playStore-release.apk susiai-master-release.apk
```
Comment (iamareebjamal) [inline exec/apk-gen.sh]: Too repetitive, see other scripts how we do it

File: exec/apk-gen.sh [hunk 4]
```diff
@@ -37,6 +37,28 @@ then
...
+    # Remove unwanted apk files
+    rm app-fdroid-release-unsigned.apk
+    rm app-playStore-release-unaligned.apk
+    rm susi-ai-master-app-playStore-release-unaligned.apk
+
+    # Remove the json files
+    rm fdroidDebug-output.json
+    rm fdroidRelease-output.json
+    rm playStoreDebug-output.json
+    rm playStoreRelease-output.json
```
Comment (iamareebjamal) [inline exec/apk-gen.sh]: No need for rm, just don't copy

File: exec/apk-gen.sh (hunk 3 above)
Comment (atm1504) [reply exec/apk-gen.sh]: We can do using the loop, but there is an issue. We won't be able to rename it in the way that Mario wants. He gave a list that cannot be renamed using the loop. 
I went through the script present in [phimpme-android](https://github.com/fossasia/phimpme-android/blob/development/scripts/update-apk.sh) which renames via loop(keeping the end portion completely same, which is different from the one expected here)

Comment (iamareebjamal) [reply exec/apk-gen.sh]: What's not possible?

Comment (atm1504) [reply exec/apk-gen.sh]: In badge-magic renaming of the files are done by passing it through a for loop sentence.
Screenshot 2019-07-02 at 7 14 28 AM

In that case, only the initial part of the name ( like the app name) can be added. But what we want here is that the entire apk to be renamed and the words like playStore, app is removed from the name.

Comment (atm1504) [reply exec/apk-gen.sh]: The similar renaming has also been done in pslab-android, there also it is done manually because of completely custom naming.

Comment (iamareebjamal) [reply exec/apk-gen.sh]: You can still do that. First remove playStore-debug and release and then do renaming. No issue

Comment (atm1504) [reply exec/apk-gen.sh]: okk, I will do it

**Comment by atm1504** [comment]
@iamareebjamal please review it. It's working (I tested for f-droid version only).

**Comment by atm1504** [comment]
@iamareebjamal please review it.

**Comment by iamareebjamal** [comment]
Make the changes listed here

**Comment by atm1504** [comment]
> Make the changes listed here [fossasia/open-event-organizer-android#1770](https://github.com/fossasia/open-event-organizer-android/pull/1770)

@iamareebjamal 
I went through the changes that u did there. But I faced an issue while making these changes in susi. There is no ```travis.yml``` file. The scripts are executed using the ```circle.yml``` file.  
What should I do now? Shall I make the following changes in the ```apk-gen.sh``` file:
Current status after I removed copying of json files:
```	\cp -r ../app/build/outputs/apk/fdroid/*/**.apk .```
```\cp -r ../app/build/outputs/apk/playStore/*/**.apk .```

Changes present in fossasia/open-event-organizer-android#1770
```find ../app/build/outputs -type f -name '*.apk' -exec cp -v {} . \;```
```find ../app/build/outputs -type f -name '*.aab' -exec cp -v {} . \;```

Please provide your views

**Comment by iamareebjamal** [comment]
If you saw the diff, you'll know same thing was present in the script in that repo. That should answer your question

**Comment by atm1504** [comment]
@iamareebjamal please review it.

**Comment by atm1504** [comment]
@iamareebjamal please review it

File: exec/apk-gen.sh [hunk 5]
```diff
@@ -33,10 +30,37 @@ then
...
+    for file in *.apk;do
+        if [[ "$file" != *"master"* ]];then
+            mv $file susiai-dev-${file%%}
+        fi
+    done
+
+    # Remove unwanted apk files
+    rm susiai-dev-fdroid-release-unsigned.apk
+    rm susiai-dev-release-unaligned.apk
+    rm susiai-master-release-unaligned.apk
```
Comment (iamareebjamal) [inline exec/apk-gen.sh]: You still haven't seen other scripts

**Comment by atm1504** [comment]
@iamareebjamal please review it. The script is now similar to ```/scripts/update-apk.sh``` present in open-event-attendee-android app

**Comment by iamareebjamal** [comment]
You made changes from https://github.com/fossasia/open-event-organizer-android/pull/1770 in a single file and ignored all others. Now master build is failing

There's a reason I gave you a reference PR

**Pull request merged**